pkg/service/controller/servingcert/secret_updating_controller.go

```diff
@@ -234,7 +236,13 @@ func (sc *ServiceServingCertUpdateController) syncSecret(key string) error {
 	dnsName := secret.Annotations[ServiceNameAnnotation] + "." + secret.Namespace + ".svc"
 	fqDNSName := dnsName + "." + sc.dnsSuffix
 	certificateLifetime := 365 * 2 // 2 years
-	servingCert, err := sc.ca.MakeServerCert(sets.NewString(dnsName, fqDNSName), certificateLifetime)
+	servingCert, err := sc.ca.MakeServerCert(
+		sets.NewString(dnsName, fqDNSName),
+		certificateLifetime,
+		extensions.ServiceServerCertificateExtension(&kapi.Service{
+			ObjectMeta: kapi.ObjectMeta{UID: types.UID(secret.Annotations[ServiceUIDAnnotation])},
```

Comment:
To state it a different way.  By signing for this information, you're asserting that the information is true, you're asserting that no one can trick the signer into signing arbitrary data, and you need to be sure that if this cert is stolen the information it asserts cannot be used by someone else without significant control of the rest of system (say owning DNS and IP routing).